app/src/main/java/com/glia/exampleapp/ExampleCustomMessageAdapter.java

```diff
@@ -0,0 +1,87 @@
+package com.glia.exampleapp;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.graphics.Typeface;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.res.ResourcesCompat;
+
+import com.glia.androidsdk.chat.ChatMessage;
+import com.glia.widgets.UiTheme;
+import com.glia.widgets.chat.adapter.CustomMessageAdapter;
+import com.glia.widgets.chat.adapter.holder.CustomMessageViewHolder;
+import com.glia.widgets.chat.adapter.holder.WebViewViewHolder;
+
+import org.json.JSONException;
+
+public class ExampleCustomMessageAdapter extends CustomMessageAdapter {
+    private static final int WEB_VIEW_TYPE = 1;
+    private static final int NATIVE_VIEW_TYPE = 2;
```

Comment:
Does it mean that this example adapter can render metadata using two additional message types: web view and custom native? So it's not obligatory for the integrator to implement two types, it could be any - just one, or more than two, right? If so, then it may be worth adding comments to this example adapter.

Reply:
Integrators can provide as many types as they need.